sig-and-wg/sig-core/proposals/generating-changelog-with-pr-labels.md

```diff
@@ -0,0 +1,103 @@
+- All changes are grouped by selected labels
+
+Desired output would look like this:
+    - [example GitHub releases](https://github.com/lerna/lerna-changelog/releases)
+    - [example CHANGELOG.md](https://github.com/lerna/lerna-changelog/blob/master/CHANGELOG.md)
+
+### Labels
+
+For now we don't introduce any new labels. We reuse `area/` labels from [the accepted proposal](https://github.wdf.sap.corp/SAP-CP-Extension-Factory/community/blob/master/sig-and-wg/wg-github-issues-migration/proposals/github-issues-labels-proposal.md). This means that all changes are grouped by area (e.g. `installation`, `security`...).
+
+**Example:**
+
+> ### Installation
+> - Sample change 1
+> - Sample change 2
+>
+> ### Security
+> - Sample change 1
+> - Sample change 2
+
+In future we can introduce nested grouping for changelog (first level: change type, second one: area of the change). Change type will marked with additional prefixed labels. Choosing right prefix will be discussed later.
+
+**Example:**
+
+> ### Bug fixes
+> **Installation**
+> - Sample change 1
+> - Sample change 2
+>
+> **Security**
+> - Sample change 1
+> - Sample change 2
+>
+> ### Features
+> **Installation**
+> - Sample change 1
+> - Sample change 2
+>
+> **Security**
+> - Sample change 1
+> - Sample change 2
+
+## Pros
+
+- One place to define general changes in pull request: pull request title, which is usually prefilled with commit message, if there is just only one commit
+- Selecting type of change is a no-brainer: contributor expands the list of available PR labels and chooses right category.
+- We can filter unlabeled pull requests ([link](https://github.com/kyma-project/kyma/issues?q=is%3Aopen+is%3Apr+no%3Alabel)) and easily add labels to them to generate complete changelog. Also, it's easy to see the full list of PRs to check if they are named correctly (and adjust titles - we can do that even for closed pull requests)
+- It enforces user to make a single type of change in a pull request
+- There are many tools available for generating changelog from pull requests labels.
+
+## Cons
+
+- More detailed changes in changelog. For example, when there is a bigger feature, which needs multiple PRs, it will be visible as multiple bullet points.
+- Pull request authors and reviewers can forget about adding a label to categorize the change. Anyway, even closed pull requests can be edited before generating the changelog. 
+
+## Tools
+
+The changelog can be generated during CI build job. There are a lot of tools for that. 
+
+#### [Lerna Changelog](https://github.com/lerna/lerna-changelog)
+
+- Very simple tool
+- Generates changelog with PR links and authors to standard output ([example](https://github.com/lerna/lerna-changelog/releases)) - we can easily pipe it to a CHANGELOG.md file and as GitHub release
```

Comment:
``Generates the changelog with the PR links and authors to standard output. See the example [here](https://github.com/lerna/lerna-changelog/releases). It can be easily piped to a `CHANGELOG.md` file and as a GitHub release.``